### PR TITLE
Add sourcelink support to SDK projects

### PR DESCRIPTION
**Pull request by eddynaka**
Fix Issue #1760 .

## Changes
Adding common properties to _nupkg.props;
Adding package reference to Microsoft.SourceLink.GitHub

### Checklist
- [ ] I ran Unit Tests locally.
- [X] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

**Comment by eddynaka** [comment]
Ok! I will add those ASAP! Give me some minutes!

Thanks for showing that
________________________________

File: .props/_Nupkg.props [hunk 1]
```diff
@@ -39,6 +39,21 @@
...
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+  </PropertyGroup>
+    
+  <PropertyGroup Condition="'$(BuildServer)' == 'true'">
+    <!-- These properties are required for SourceLink and should only be set to true on a CI/Build Server -->
+    
+    <!-- indicate that the build executes on a build/CI server -->
+    <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
+    <!-- mapping all source paths included in the project outputs to deterministic values -->    
+    <DeterministicSourcePaths>true</DeterministicSourcePaths>
```
Comment (clairernovotny) [inline .props/_Nupkg.props:56]: No need to set this manually, this should be set by the targets

File: BASE/src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj [hunk 2]
```diff
@@ -41,8 +41,12 @@
     </PackageReference>
   </ItemGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition=" $(OS) == 'Windows_NT'">
     <!--Build Infrastructure-->
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0">
```
Comment (clairernovotny) [inline BASE/src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj:46]: The SourceLink reference should be added on all OS's

File: BASE/src/ServerTelemetryChannel/TelemetryChannel.csproj [hunk 3]
```diff
@@ -48,8 +48,12 @@
     </PackageReference>
   </ItemGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition=" $(OS) == 'Windows_NT'">
     <!--Build Infrastructure-->
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0">
```
Comment (clairernovotny) [inline BASE/src/ServerTelemetryChannel/TelemetryChannel.csproj:53]: The SourceLink reference should be added on all OS's

**Comment by clairernovotny** [comment]
I would not recommend adding SourceLink to each project individually. It's a pain to then update. Put the reference into a Directory.Build.props file

File: .props/_Nupkg.props [hunk 4]
```diff
@@ -39,6 +39,21 @@
...
     <PackageTags>Analytics Azure ApplicationInsights Telemetry Monitoring SDK</PackageTags>
+    
+    <!-- Publish the repository URL in the built .nupkg (in the NuSpec <Repository> element) -->
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+
+    <!-- Embed source files that are not tracked by the source control manager in the PDB -->
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+  </PropertyGroup>
+    
+  <PropertyGroup Condition="'$(BuildServer)' == 'true'">
```
Comment (clairernovotny) [inline .props/_Nupkg.props:50]: This should be at Directory.Build.props

**Comment by clairernovotny** [comment]
A few things should be added to a Directory.Build.targets as well:

https://github.com/clairernovotny/DeterministicBuilds

**Pull request closed without merging**